### PR TITLE
scipoptsuite 8.0.3 (new formula)

### PR DESCRIPTION
**Pull request by jurgen-lentz**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/scipoptsuite.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Documentation: https://docs.brew.sh/Formula-Cookbook
+#                https://rubydoc.brew.sh/Formula
+# PLEASE REMOVE ALL GENERATED COMMENTS BEFORE SUBMITTING YOUR PULL REQUEST!
```
Comment (SMillerDev) [inline Formula/scipoptsuite.rb:3]: Do this

File: Formula/scipoptsuite.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  depends_on "cppad"
+  depends_on "boost"
+  depends_on "gmp"
+  depends_on "bison"
+  depends_on "flex"
+  depends_on "gnuplot"
+
+  def install
+    cmake_args = std_cmake_args + %w[
+      -DCMAKE_BUILD_TYPE=Release
```
Comment (SMillerDev) [inline Formula/scipoptsuite.rb:24]: This is already a standard argument

File: Formula/scipoptsuite.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+  test do
+    system "cmake", "-B", "build", "-S", "scipoptsuite/scip/examples/Queens", "-DCMAKE_BUILD_TYPE=Release"
+    system "cmake", "--build", "build", "--parallel", "${CPU_COUNT}"
+    system "./build/queens", "5"
+
+    system "scip", "--version"
+
+    # Verifies that dependencies are properly linked
+    system "scip", "--version", "|", "grep", "-iE", "'CppAD\s+[0-9]+'"
```
Comment (SMillerDev) [inline Formula/scipoptsuite.rb:55]:
```suggestion
    output = shell_output("scip --version")
    assert_match /CppAD\s+[0-9]+/, output
```

File: Formula/scipoptsuite.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      -DPARASCIP=ON
+      -DPAPILO=ON
+      -DSOPLEX=ON
+      -DGCG=ON
+      -DZIMPL=ON
+      -DBOOST=ON
+      -DGMP=ON
+      -DQUADMATH=ON
+      -DIPOPT=ON
+      -DIPOPT_DIR="${PREFIX}"
```
Comment (SMillerDev) [inline Formula/scipoptsuite.rb:30]: Shouldn't this be 
```suggestion
      -DIPOPT_DIR="#{prefix}"
```

File: Formula/scipoptsuite.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      -DZLIB=ON
+      -DREADLINE=OFF
+      -DSYM=bliss
+      -DEXPRINT=cppad
+      -DCLIQUER=OFF
+    ]
+
+    system "cmake", "-B", "scipoptsuite-build", "-S", ".", *cmake_args
+    system "cmake", "--build", "scipoptsuite-build"
+    system "cmake", "--install", "scipoptsuite-build", "--prefix", "${PREFIX}"
```
Comment (SMillerDev) [inline Formula/scipoptsuite.rb:40]: Where does it install without the prefix specification?

File: Formula/scipoptsuite.rb [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    system "cmake", "-B", "build", "-S", "scipoptsuite/scip/examples/Queens", "-DCMAKE_BUILD_TYPE=Release"
+    system "cmake", "--build", "build"
+    system "./build/queens", "5"
+
+    system "scip", "--version"
+
+    # Verifies that dependencies are properly linked
+    output = shell_output("scip --version")
+    assert_match "CppAD\s+[0-9]+", output
+    output = shell_output("scip --version")
```
Comment (SMillerDev) [inline Formula/scipoptsuite.rb:53]: No need to run this every time, it'll still be the same output

File: Formula/scipoptsuite.rb (hunk 5 above)
Comment (jurgen-lentz) [reply Formula/scipoptsuite.rb]: Ok, I think this is also not needed. Is there a way to use all available cores for building?

Comment (SMillerDev) [reply Formula/scipoptsuite.rb]: The cores are automatically scaled. But what does that have to do with `--prefix`

Comment (jurgen-lentz) [reply Formula/scipoptsuite.rb]: Nothing. I already deleted ```--prefix```.

File: Formula/scipoptsuite.rb [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  depends_on "cppad"
+  depends_on "flex"
+  depends_on "gmp"
+  depends_on "gnuplot"
+  depends_on "ipopt"
+  depends_on "tbb"
+  depends_on "zlib"
+
+  def install
+    cmake_args = std_cmake_args + %w[
```
Comment (SMillerDev) [inline Formula/scipoptsuite.rb:20]:
```suggestion
    cmake_args = std_cmake_args + %W[
```

File: Formula/scipoptsuite.rb [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+  end
+
+  test do
+    system "cmake", "-B", "build", "-S", "scipoptsuite/scip/examples/Queens", "-DCMAKE_BUILD_TYPE=Release"
+    system "cmake", "--build", "build"
+    system "./build/queens", "5"
+
+    system "scip", "--version"
+
+    # Verifies that dependencies are properly linked
```
Comment (SMillerDev) [inline Formula/scipoptsuite.rb:50]: Why not compile something that links against these?

Comment (jurgen-lentz) [reply Formula/scipoptsuite.rb:50]: I used the same procedure that the team is using for testing the Scipoptsuite on conda.

Comment (SMillerDev) [reply Formula/scipoptsuite.rb:50]: I don't know the rules for coda tests. But brew's rules state that a version check isn't enough

Comment (jurgen-lentz) [reply Formula/scipoptsuite.rb:50]: Ok, thanks. Then I will work on the test.

**Comment by jurgen-lentz** [comment]
@SMillerDev how does it look to you?

File: Formula/scipoptsuite.rb [hunk 9]
```diff
@@ -0,0 +1,50 @@
+class Scipoptsuite < Formula
+  desc "Mixed Integer Programming (MIP) solver and Branch-and-Cut-and-Price Framework"
+  homepage "https://www.scipopt.org/"
+  url "https://www.scipopt.org/download/release/scipoptsuite-8.0.3.tgz"
+  sha256 "5ad50eb42254c825d96f5747d8f3568dcbff0284dfbd1a727910c5a7c2899091"
+  license all_of: ["Apache-2.0", "LGPL-3.0-or-later"]
+
+  depends_on "bison@2.7"
```
Comment (chenrui333) [inline Formula/scipoptsuite.rb:8]: Any issue to support latest bison?

File: Formula/scipoptsuite.rb [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+  depends_on "boost"
+  depends_on "cmake"
```
Comment (chenrui333) [inline Formula/scipoptsuite.rb:10]: should be build dep?

File: Formula/scipoptsuite.rb [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+  depends_on "gmp"
+  depends_on "gnuplot"
+  depends_on "ipopt"
+  depends_on "tbb"
+
+  uses_from_macos "flex"
+  uses_from_macos "zlib"
+
+  def install
+    cmake_args = std_cmake_args + %W[
```
Comment (chenrui333) [inline Formula/scipoptsuite.rb:21]:
```suggestion
    cmake_args = %W[
```

File: Formula/scipoptsuite.rb [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+      -DIPOPT=ON
+      -DIPOPT_DIR="#{prefix}"
+      -DZLIB=ON
+      -DREADLINE=OFF
+      -DSYM=bliss
+      -DEXPRINT=cppad
+      -DCLIQUER=OFF
+    ]
+
+    system "cmake", "-B", "scipoptsuite-build", "-S", ".", *cmake_args
```
Comment (chenrui333) [inline Formula/scipoptsuite.rb:39]: let's separate args vs std_cmake_args
```suggestion
    system "cmake", "-B", "scipoptsuite-build", "-S", ".", *cmake_args, *std_cmake_args
```

**Comment by chenrui333** [comment]
left couple of comments, overall, looks good. Thanks!

File: Formula/scipoptsuite.rb (hunk 9 above)
Comment (jurgen-lentz) [reply Formula/scipoptsuite.rb:8]: Not really. I will change it.

File: Formula/scipoptsuite.rb (hunk 10 above)
Comment (jurgen-lentz) [reply Formula/scipoptsuite.rb]: changed it to build.

File: Formula/scipoptsuite.rb (hunk 9 above)
Comment (jurgen-lentz) [reply Formula/scipoptsuite.rb]: this is needed.

[the remark this answers is not in the document]
Comment (jurgen-lentz) [reply Formula/scipoptsuite.rb:9]: I first used ```depends_on "bison"``` but then I get the error that I should you ```uses_from_macos```. When I use ```uses_from_macos``` then the problem is that bison 2.3 is used but SCIP needs at least bison 2.4. What should I do?

[the remark this answers is not in the document]
Comment (SMillerDev) [reply Formula/scipoptsuite.rb:9]: In that case just depend on the latest bison.

[the remark this answers is not in the document]
Comment (jurgen-lentz) [reply Formula/scipoptsuite.rb:9]: Use only ```depends_on "bison@3.8"```?

[the remark this answers is not in the document]
Comment (SMillerDev) [reply Formula/scipoptsuite.rb:9]: No, `depends_on "bison"` only

File: Formula/scipoptsuite.rb [hunk 13]
```diff
@@ -0,0 +1,50 @@
+class Scipoptsuite < Formula
+  desc "Mixed Integer Programming (MIP) solver and Branch-and-Cut-and-Price Framework"
+  homepage "https://www.scipopt.org/"
+  url "https://www.scipopt.org/download/release/scipoptsuite-8.0.3.tgz"
+  sha256 "5ad50eb42254c825d96f5747d8f3568dcbff0284dfbd1a727910c5a7c2899091"
+  license all_of: ["Apache-2.0", "LGPL-3.0-or-later"]
+
+  depends_on "cmake" => :build
+  depends_on "bison"
```
Comment (SMillerDev) [inline Formula/scipoptsuite.rb:9]: Is if needed at runtime too?

File: Formula/scipoptsuite.rb [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+  depends_on "boost"
+  depends_on "cppad"
+  depends_on "gmp"
+  depends_on "gnuplot"
+  depends_on "ipopt"
+  depends_on "tbb"
+
+  uses_from_macos "flex"
```
Comment (SMillerDev) [inline Formula/scipoptsuite.rb:17]: Is this needed at runtime?

File: Formula/scipoptsuite.rb (hunk 9 above)
[the remark this answers is not in the document]
Comment (jurgen-lentz) [reply Formula/scipoptsuite.rb]: I receive the following problem ```* Dependency 'bison' is provided by macOS; please replace 'depends_on' with 'uses_from_macos'.```. When I use ```uses_from_macos``` the wrong version of bison is used.

[the remark this answers is not in the document]
Comment (SMillerDev) [reply Formula/scipoptsuite.rb]: I know, maintainers can decide to make exceptions for audits as needed, which is a better solution than depending on an outdated version.

[the remark this answers is not in the document]
Comment (jurgen-lentz) [reply Formula/scipoptsuite.rb]: Alright, thanks you.

File: Formula/scipoptsuite.rb (hunk 13 above)
Comment (jurgen-lentz) [reply Formula/scipoptsuite.rb:9]: yes, zimpl uses it at runtime.

File: Formula/scipoptsuite.rb (hunk 14 above)
Comment (jurgen-lentz) [reply Formula/scipoptsuite.rb:17]: yes, zimpl uses it at runtime.

**Comment by jurgen-lentz** [comment]
It should be able to run now.

**Review by SMillerDev — APPROVED** [review]

**Comment by jurgen-lentz** [comment]
Do I need to change something?

**Comment by jurgen-lentz** [comment]
Or shoild I rather create one Formula for Soplex, one for SCIP, one for GCG and one for ZIMPL?

**Comment by SMillerDev** [comment]
That seems easier to maintain than vendoring them all here so I'm in favour

**Comment by jurgen-lentz** [comment]
Alright. I will then close this MR and start to create them one by one.

**Pull request closed without merging**